torch/csrc/distributed/c10d/init.cpp

```diff
@@ -106,6 +106,28 @@ class PythonStore : public ::c10d::Store {
   }
 };
 
+// PythonhookBinder has a static function called register_comm_hook
+// that enables registering a c10d PythonCommHook object to c10d
+// reducer from Python.
+class PythonHookBinder {
```

Comment:
Why do we need this wrapper class? Can't we just have a standalone `register_comm_hook` function and bind that to python?